pkg/build/api/types.go

```diff
@@ -397,6 +397,9 @@ const (
 	// availability of a new version of an image
 	ImageChangeBuildTriggerType           BuildTriggerType = "ImageChange"
 	ImageChangeBuildTriggerTypeDeprecated BuildTriggerType = "imageChange"
+
+	// ConfigChangeBuildTriggerType will trigger a build on an initial build config creation
+	ConfigChangeBuildTriggerType BuildTriggerType = "ConfigChange"
```

Comment:
It's the result of a discussion we had regarding this feature with @bparees and @smarterclayton a few days ago. We don't yet want to implement a full config change controller as in deployments.